R/computeMode3.R

```diff
@@ -0,0 +1,26 @@
+#' computeMode(c(1,2,3,3))
+computeMode3 = function(x, ties.method = "random", na.rm = TRUE) {
+
+  tm = switch(ties.method, "first" = 1L, "last" = 2L, "random" = 3L)
+
+  if (is.integer(x)) {
+    shift = -min(x) + 1L
+    if (shift > 0)
+      x = x + shift
+    mode = .Call("c_compute_mode", x, tm, PACKAGE = "BBmisc")
+    if (shift > 0)
+      mode = mode - shift
+    return(mode)
+  } else if (is.numeric(x) || is.character(x)) {
+    f = as.factor(x) # this slows everything down
```

Comment:
horribly slow

Reply:
Handling non-integer variables will require a full-featured hash table implementation in C, or in other words, some hundred lines of code to get it right.